scene/gui/label.cpp

```diff
@@ -447,12 +452,19 @@ void Label::regenerate_word_cache() {
 
 			if (current == '\n') {
 				insert_newline = true;
+				prev_insert_newline = true;
 			} else if (current != ' ') {
 				total_char_cache++;
 			}
 
 			if (i < xl_text.length() && xl_text[i] == ' ') {
-				if (line_width > 0 || last == nullptr || last->char_pos != WordCache::CHAR_WRAPLINE) {
+				if (i == 0 || prev_insert_newline) {
```

Comment:
Having read your last description about how these variables work again. I think `prev_insert_newline` here is equivalent to checking `line_width == 0`. What do you think?

Reply:
@timothyqiu yes, that's true! I didn't think of that. I've implemented the fix without the `prev_insert_newline` variable and the codebase fix now looks alot cleaner and simple. I've pushed a new commit with your suggestion, when a newline is made, i set `line_width` back to 0. The fix works better than the last fix with the local variable :) thanks!